stl/inc/ranges

```diff
@@ -4749,6 +4752,477 @@ namespace ranges {
+                    --_Inner_it._Second;
+                    break;
+                case _Variantish_state::_Nothing:
+                    _Throw_bad_variant_access();
+                }
+                return *this;
+            }
+
+            constexpr _Iterator operator--(int) requires _Deref_is_glvalue && bidirectional_range<_Base> //
+                && _Bidirectional_common<_InnerRng<_Const>> && _Bidirectional_common<_PatternBase> {
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            friend constexpr bool operator==(const _Iterator& _Left, const _Iterator& _Right) requires _Deref_is_glvalue
+                && equality_comparable<_OuterIter> && equality_comparable<_InnerIter> {
+                if (_Left._Outer_it != _Right._Outer_it) {
+                    return false;
+                }
+
+                if (_Left._Inner_it._Contains != _Right._Inner_it._Contains) {
+                    return false;
+                }
+
+                switch (_Left._Inner_it._Contains) {
+                case _Variantish_state::_Holds_first:
+                    return _Left._Inner_it._First == _Right._Inner_it._First;
+                case _Variantish_state::_Holds_second:
+                    return _Left._Inner_it._Second == _Right._Inner_it._Second;
+                case _Variantish_state::_Nothing:
+                    return true;
+                }
+
+                _STL_UNREACHABLE;
+            }
+
+            friend constexpr decltype(auto) iter_move(const _Iterator& _It) {
+                using _Rvalue_ref =
+                    common_reference_t<iter_rvalue_reference_t<_InnerIter>, iter_rvalue_reference_t<_PatternIter>>;
+                return _It._Visit_inner_it<_Rvalue_ref>(_RANGES iter_move);
+            }
+
+            friend constexpr void iter_swap(const _Iterator& _Left,
+                const _Iterator& _Right) requires indirectly_swappable<_InnerIter, _PatternIter> {
+                switch (_Left._Inner_it._Contains) {
+                case _Variantish_state::_Holds_first:
+                    switch (_Right._Inner_it._Contains) {
+                    case _Variantish_state::_Holds_first:
+                        return _RANGES iter_swap(_Left._Inner_it._First, _Right._Inner_it._First);
+                    case _Variantish_state::_Holds_second:
+                        return _RANGES iter_swap(_Left._Inner_it._First, _Right._Inner_it._Second);
+                    }
+                case _Variantish_state::_Holds_second:
+                    switch (_Right._Inner_it._Contains) {
+                    case _Variantish_state::_Holds_first:
+                        return _RANGES iter_swap(_Left._Inner_it._Second, _Right._Inner_it._First);
+                    case _Variantish_state::_Holds_second:
+                        return _RANGES iter_swap(_Left._Inner_it._Second, _Right._Inner_it._Second);
+                    }
+                }

```

Comment:
There is a bug here - if the first sub-switch `switch (_Right._Inner_it._Contains)` does nothing, it silently falls through to the second case of the outer `switch` (right line 5075).

I believe that we should:
* Give each of the sub-switches an explicit `case _Variantish_state::_Nothing: break;`
* Add `break;` after each `case` of the outer `switch`.
* Give the outer `switch` an explicit `case _Variantish_state::_Nothing: break;`.

Then this would clearly show any "nothing" states ending up at the `_Throw_bad_variant_access()` explicitly.